### PR TITLE
[LoadableByAddress] Set the correct debug scope to avoid verification…

### PR DESCRIPTION
**Pull request by dcci**
… failures.

We need to skip over AllocStackInsts as they can be hoisted and mess
up with the scope (and, FWIW, they're always skipped as part of verification
as they have no semantic value associated).

<rdar://problem/43498112>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
@adrian-prantl Adrian, I would appreciate if you can review this.

File: lib/IRGen/LoadableByAddress.cpp [hunk 1]
```diff
@@ -1491,7 +1491,19 @@ void LoadableStorageAllocation::allocateForArg(SILValue value) {
 
   assert(!ApplySite::isa(value) && "Unexpected instruction");
 
-  SILBuilderWithScope allocBuilder(&*pass.F->begin());
+  // Find the first non-alloc stack and use its scope when creating
```
Comment (adrian-prantl) [inline lib/IRGen/LoadableByAddress.cpp]: `non-AllocStackInst` or `non-alloc_stack`?

File: lib/IRGen/LoadableByAddress.cpp [hunk 2]
```diff
@@ -1491,7 +1491,19 @@ void LoadableStorageAllocation::allocateForArg(SILValue value) {
...
+  // the new SILBuilder. AllocStackInst(s) can be hoisted and mess
+  // up with scopes.
```
Comment (adrian-prantl) [inline lib/IRGen/LoadableByAddress.cpp]: That's true, but the real reason is:
An alloc_stack does not directly cause any code to be generated. The location of an alloc_stack carries information about the source variable; it doesn't matter where in the instruction stream the alloc_stack is located.

**Review by adrian-prantl — APPROVED** [review]

**Comment by dcci** [comment]
Updated the comment. thanks!

**Comment by dcci** [comment]
@swift-ci test and merge

**Comment by dcci** [comment]
@swift-ci test and merge

**Comment by shahmishal** [comment]
@swift-ci test and merge

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
@swift-ci please test and merge

**Comment by dcci** [comment]
@swift-ci please test and merge

**Pull request merged**